internal/kafka/internal/services/quota/ams_quota_service.go

```diff
@@ -112,12 +113,12 @@ func (q amsQuotaService) getAvailableBillingModelFromKafkaInstanceType(externalI
 	return billingModel, nil
 }
 
-func (q amsQuotaService) ReserveQuota(kafka *dbapi.KafkaRequest, instanceType types.KafkaInstanceType) (string, *errors.ServiceError) {
+func (q amsQuotaService) ReserveQuota(kafka *dbapi.KafkaRequest, instanceType types.KafkaInstanceType, sizeRequired int, kafkaConfig *config.KafkaConfig) (string, *errors.ServiceError) {
 	kafkaId := kafka.ID
 
 	rr := newBaseQuotaReservedResourceResourceBuilder()
 
-	bm, err := q.getAvailableBillingModelFromKafkaInstanceType(kafka.OrganisationId, instanceType)
+	bm, err := q.getAvailableBillingModelFromKafkaInstanceType(kafka.OrganisationId, instanceType, sizeRequired)
```

Comment:
Is this change verified at the AMS api?

Reply:
no

Reply:
Verified:

1. Checked quota available: it was `consumed:1, allowed:3`
2. Reserved a resource with count 2
3. Checked quota available: it was `consumed:3, allowed:3`
4. Deleted the resource 
3. Checked quota available: it was `consumed:1, allowed:3`

Reply:
verification of the same logic was verified by Massimo